gobblin-binary-management/src/main/java/org/apache/gobblin/binary_creation/OrcTestTools.java

```diff
@@ -0,0 +1,408 @@
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.ShortWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+
+import static org.apache.gobblin.binary_creation.AvroTestTools.*;
+
+
+// A class that examines ORC-Format file in Purger Integration test.
+@Slf4j
+public class OrcTestTools extends DataTestTools<OrcTestTools.OrcRowIterator, TypeInfo> {
+
+  /**
+   *
+   * @param expected
+   * @param observed
+   * @param allowDifferentOrder ORC tools will not use this parameter currently.
+   * @param blacklistRecordFields ORC tools will not use this parameter currently.
+   * @return If two sets of files are identical.
+   * Note that there might be an ordering issue in this comparison method. When one is drafting an ORC integration
+   * test, try to name all json files differently.
+   */
+  @Override
+  public boolean checkSameFilesAndRecords(TreeMap<String, OrcRowIterator> expected,
+      TreeMap<String, OrcRowIterator> observed, boolean allowDifferentOrder, Collection<String> blacklistRecordFields,
+      boolean allowDifferentSchema) {
+    Iterator<String> keys1 = expected.navigableKeySet().iterator();
+    Iterator<String> keys2 = observed.navigableKeySet().iterator();
+
+    return compareIterators(keys1, keys2, (key1, key2) -> {
+      // ORC file doesn't have extension by Linkedin's convention.
+      if (!removeExtension(key1).equals(key2)) {
+        log.error(String.format("Mismatched files: %s and %s", key1, key2));
+        return false;
+      }
+
+      OrcRowIterator it1 = expected.get(key1);
+      OrcRowIterator it2 = observed.get(key2);
+
+      if (!it1.getTypeInfo().equals(it2.getTypeInfo())) {
+        log.error(String.format("Mismatched Typeinfo: %s and %s", key1, key2));
+        return false;
+      }
+
+      boolean result = true;
+      while (it1.hasNext()) {
+        if (!it2.hasNext()) {
+          return false;
+        }
+
+        result &= compareJavaRowAndOrcStruct(((AvroRow) it1.next()).getRow(), (OrcStruct) it2.next());
+      }
+      return result;
+    });
+  }
+
+  /**
+   * Given the fact that we couldn't access OrcStruct easily, here uses the hacky way(reflection)
+   * to go around access modifier for integration test purpose only.
+   * @param realRow A row containing a list of Java objects.
+   * @param struct An {@link OrcStruct} which essentially is a list of {@link Writable} objects.
+   */
+  private boolean compareJavaRowAndOrcStruct(Object realRow, OrcStruct struct) {
+    boolean isIdentical = true;
+    ArrayList<Object> javaObjRow = (ArrayList) realRow;
+
+    try {
+      Field objectArr = OrcStruct.class.getDeclaredField("fields");
+      objectArr.setAccessible(true);
+      Object[] dataArr = (Object[]) objectArr.get(struct);
+
+      int index = 0;
+      for (Object dataField : dataArr) {
+        if (dataField instanceof OrcStruct) {
+          isIdentical &= compareJavaRowAndOrcStruct(javaObjRow.get(index), (OrcStruct) dataField);
```

Comment:
Short-circuit instead of &=. Not really a problem for test code, but I want to avoid this being propagated through copy and paste into more performance critical paths.